CheoR/pack-it-up
# Cr refactors

# Description

When user deletes a Move, any boxes linked to the move also delete. However items linked to the boxes did no delete.
Now when user deletes a move, the items linked to the boxes are fetched and deleted first before the Move and linked boxes are deleted.

Closes #83 

## Type of change

Please delete options that are not relevant.


# Testing Instructions

`git fetch --all`
`git checkout cr-refactor`
`npm start`

In another tab, cd into `api` and run

`json-server -p 8088  database.json`

1. Create Moves, Boxes, and Items.
2. Delete a Move that has linked Boxes and Items.
3. All linked Boxes and Items should also be removed from the db.

# Checklist:
